src/morale.h

```diff
@@ -73,122 +81,120 @@ enum morale_type : int {
     NUM_MORALE_TYPES
 };
 
-// Morale multiplier
-struct morale_mult {
-    morale_mult(): good( 1.0 ), bad( 1.0 ) {}
-    morale_mult( double good, double bad ): good( good ), bad( bad ) {}
-    morale_mult( double both ): good( both ), bad( both ) {}
-
-    double good;    // For good morale
-    double bad;     // For bad morale
-
-    morale_mult operator * ( const morale_mult &rhs ) const {
-        return morale_mult( *this ) *= rhs;
-    }
-
-    morale_mult &operator *= ( const morale_mult &rhs ) {
-        good *= rhs.good;
-        bad *= rhs.bad;
-        return *this;
-    }
-};
-
-inline double operator * ( double morale, const morale_mult &mult )
+class player_morale
 {
-    return morale * ( ( morale >= 0.0 ) ? mult.good : mult.bad );
-}
-
-inline double operator * ( const morale_mult &mult, double morale )
-{
-    return morale * mult;
-}
+    public:
+        player_morale() :
+            level_is_valid( false ),
+            took_prozak( false ),
+            stylish( false ),
+            super_fancy_bonus( 0 ) {};
+        player_morale( player_morale && ) = default;
+        player_morale( const player_morale & ) = default;
+        player_morale &operator =( player_morale && ) = default;
+        player_morale &operator =( const player_morale & ) = default;
+
+        /** Adds morale to existing or creates one */
+        void add( morale_type type, int bonus, int max_bonus = 0, int duration = MINUTES( 6 ),
+                  int decay_start = MINUTES( 3 ), bool capped = false, const itype *item_type = nullptr );
+        /** Adds permanent morale to existing or creates one */
+        void add_permanent( morale_type type, int bonus, int max_bonus = 0,
+                            bool capped = false, const itype *item_type = nullptr );
+        /** Returns bonus from specified morale */
+        int has( morale_type type, const itype *item_type = nullptr ) const;
+        /** Removes specified morale */
+        void remove( morale_type type, const itype *item_type = nullptr );
+        /** Clears up all morale points */
+        void clear();
+        /** Returns overall morale level */
+        int get_level() const;
+        /** Ticks down morale counters and removes them */
+        void decay( int ticks = 1 );
+        /** Displays morale screen */
+        void display();
+
+    protected:
+        friend class player;
```

Comment:
Usually, I'd rather use inheritance to make protected methods accessible than modify the initial class to make the tests happy. Either way, this coupling with the player is supposed to be temporary.